L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc

```diff
@@ -1000,6 +1020,41 @@ std::pair<unsigned int, unsigned int> L1TCorrelatorLayer1Producer::totAndMax(
   }
   return std::make_pair(ntot, nmax);
 }
+//do a quick histogram vertexing since the current collections only output the leading vertex
+void L1TCorrelatorLayer1Producer::doVertexings(std::vector<float> &pvdz) const {
+  int lNBins = int(40. / vtxRes_);
+  lNBins *= 3;
+  std::unique_ptr<TH1F> h_dz(new TH1F("h_dz", "h_dz", lNBins, -20, 20));
+  for (const auto r : event_.pfinputs) {
+    const auto &reg = r.region;
+    for (const auto &p : r.track) {
```

Comment:
This won't work as expected, because it will only use the tracks in the regions considered by this module, so you'll get a barrel vertex made with just barrel tracks and an endcap vertex made just with endcap tracks.
You should loop instead on the sectors in event_.decoded.track, which would give you all the tracks (and you can skip the isFiducial check later)

That said, I would very much prefer if we could just read a collection of multiple PVs from GTT rather than resurrect this hacked in-house vertex finding.

Reply:
agreed that we shouldn't use this I added it as an option in the config that is currently set to false. I don't see any drop in efficiency. I also don't observe a drop in efficiency with 1 vertex.